### ISSUE TITLE
unable to get accessibility info using text fields with hints

### ISSUE THREAD

**Issue by JigneshWorld** [flutter/flutter#119326]
## Steps to Reproduce

I have been using [maestro](https://maestro.mobile.dev/) for testing my apps and using accessibility info in `maestro studio` but this can be accessed in normal `adb` `uiautomatorviewer` or any other tools on top of that.

1. Try Flutter Gallery app built with Flutter 3.3 
2. Try Flutter Gallery app built with Flutter 3.7
3. Observe the difference as shown in these videos 

**Expected results:** 
This should work the way it is working in Flutter 3.3
OR 
We should have this as a **breaking change** with a workaround. 

https://user-images.githubusercontent.com/7347854/215075106-4f013baa-702a-4d40-933b-7470dca82327.mp4

**Actual results:** 
It's not passing accessibility info using Text Fields with hints.

https://user-images.githubusercontent.com/7347854/215075091-466bb221-ba72-44b2-a172-6c31631e5d6f.mp4

**Comment by darshankawar** [comment]
Thanks for the detailed report. Using steps provided, seeing the same behavior as reported.
Not completely sure if this could be related to https://github.com/flutter/flutter/issues/114511 so I'll let the team share their thoughts if this is a regression and whether this need to be documented for the changes seen in latest stable version.

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.7.0, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.7.0 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b06b8b2710 (28 hours ago), 2023-01-23 16:55:55 -0800
    • Engine revision b24591ed32
    • Dart version 2.19.0
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.7.0-29.0.pre.26, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.7.0-29.0.pre.26 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 15cd00f1ed (2 hours ago), 2023-01-29 21:58:05 -0500
    • Engine revision a7bb0e4108
    • Dart version 3.0.0 (build 3.0.0-179.0.dev)
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by JigneshWorld** [comment]
Tried a few different things to expose this to the system but I can't able to get any workaround for this. 

Here is a screenshot from `updated code` and `uiautomatorviewer` too.
![Screenshot 2023-01-31 at 12 26 14 PM](https://user-images.githubusercontent.com/7347854/215688443-32735388-8dcb-49ef-a7d3-408b486aa16a.png)

@darshankawar @chunhtai  Any workaround?
If there is no possible workaround then I think we should assign this some higher priority or maybe regression.

**Comment by chunhtai** [comment]
The label is set as https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#getHintText()

Can you use that to query the textfield?

**Comment by JigneshWorld** [comment]
As per my understanding of the toolings around this, this is represented as a `text` in android `uiautomatorviewer`. 

I am not sure about what other way you would like me to query this text, if you please share any reference steps or directions then I would be happy to share those results. 

Attaching some screenshots from `uiautomatorviewer` if that can be helpful. 

Before Flutter 3.7:

![Screenshot 2023-02-02 at 11 50 59 AM](https://user-images.githubusercontent.com/7347854/216253100-02f49f53-3dfe-4c20-8483-412e621cb894.png)


After Flutter 3.7:

![Screenshot 2023-02-02 at 11 49 17 AM](https://user-images.githubusercontent.com/7347854/216253607-4c5f5d21-6fb9-42e8-ba76-1ef54453f93c.png)

**Comment by chunhtai** [comment]
The change is introduced in the 3.7 release, previously the label was set to the text which causes issue when using voice access. It is now set to hint. I think the issue is that the UIAutomator does not have api to query the hint text.

This should be an android issue

**Comment by chunhtai** [comment]
filed internal issue b/267640877

**Comment by JigneshWorld** [comment]
Thanks, @chunhtai for pointing in this direction. 
Just opened the issue in the `maestro` tool to access this info.

**Comment by JigneshWorld** [comment]
As this is a change in the new flutter version, @chunhtai does it consider a `breaking change` for teams who are using this?

**Comment by dmitry-zaitsev** [comment]
Author of [Maestro](https://github.com/mobile-dev-inc/maestro) here.

It appears that [Hint](https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#getHintText()) property is not populated either, I have verified that by both running `adb shell uiautomator dump` as well as programmatically accessing this information - the value is empty.

**Comment by chunhtai** [comment]
Hi @dmitry-zaitsev Thanks for reply.

> It appears that [Hint](https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#getHintText()) property is not populated either, I have verified that by both running adb shell uiautomator dump as well as programmatically accessing this information - the value is empty (tested on Flutter 3.7.1)

Not populated, as is the field not appear in the dump at all, or the field exist but the value is empty.

Also what Android version are you using for testing? The code currently only setHintText for Android 9 and above.

**Comment by chunhtai** [comment]
> The field does not appear in the dump.

This is what i observe as well, I think UIAutomator does not query the field at all, which i filed an internal issue against.

May I know how do you programmatically access it? can you share a code snippet?

**Comment by dmitry-zaitsev** [comment]
Here is how we do it: https://github.com/mobile-dev-inc/maestro/blob/main/maestro-android/src/androidTest/java/dev/mobile/maestro/ViewHierarchy.kt#L96

We copied implementation from UIAutomator's package-private method. In the snippet above we are not querying the `hint` field, but it can be done simply by adding:

```
serializer.attribute("", "hint", safeCharSeqToString(node.hintText))
```

**Comment by SebastianKlaiber** [comment]
Are there any updates on the issue? Many of our tests are no longer running.

**Comment by yasht01** [comment]
Any updates to this issue? This issue still persists on `3.7.7`
